### PR TITLE
Remove data object slicing from simulation mode

### PR DESCRIPTION
**Pull request by ottomattas**

### CODE REVIEW

**Comment by ottomattas** [comment]
Currently, the full dependencies test fails with:
`E   ValueError: numpy.ndarray size changed, may indicate binary incompatibility. Expected 88 from C header, got 80 from PyObject`

Can you help me figure this out @J535D165 ?

**Comment by ottomattas** [comment]
The error looks to be unrelated as is also present for other PRs (e.g https://github.com/asreview/asreview/runs/3535156837)

**Review by J535D165 — APPROVED** [review]

**Comment by J535D165** [comment]
This PR removes the option to simulate with partially labeled datasets. Slice your dataset with your reference manager of choice before simulating. 

No @ottomattas, there is no issue yet. maybe good to create one. There is a comment of a TensorFlow dev here https://github.com/asreview/asreview/pull/711#issuecomment-912823615. 

Thanks for the implementation. @Rensvandeschoot, approve? Do you expect issues?

File: tests/test_simulate.py [hunk 1]
```diff
@@ -195,36 +194,6 @@ def test_classifiers():
...
-                n_prior_excluded=1,
-                prior_idx=None,
-                state_checker=check_partial_state)
-
-
-@pytest.mark.xfail(
-    raises=ValueError,
-    reason="prior_idx not available for partly labeled data"
-)
-def test_partial_simulation_2(tmpdir):
```
Comment (J535D165) [inline tests/test_simulate.py:216]: Is there an informative error now if the user uploads a partially labeled dataset? What error is raised?

**Review by Rensvandeschoot — APPROVED** [review]

File: tests/test_simulate.py (hunk 1 above)
Comment (ottomattas) [reply tests/test_simulate.py:216]: There is indeed a ValueError thrown by `simulate.py`:
https://github.com/asreview/asreview/blob/fee139f2dbc21689488ec59467fbe73523576213/asreview/review/simulate.py#L89

For verification, I ran a test with our representative demo dataset [generic_partial_labels.csv](https://github.com/asreview/asreview/blob/fee139f2dbc21689488ec59467fbe73523576213/tests/demo_data/generic_partial_labels.csv) :
```
% asreview simulate tests/demo_data/generic_partial_labels.csv --state_file myreview.h5 

            _____ _____            _
     /\    / ____|  __ \          (_)
    /  \  | (___ | |__) |_____   ___  _____      __
   / /\ \  \___ \|  _  // _ \ \ / / |/ _ \ \ /\ / /
  / ____ \ ____) | | \ \  __/\ V /| |  __/\ V  V /
 /_/    \_\_____/|_|  \_\___| \_/ |_|\___| \_/\_/

---------------------------------------------------------------------------------
|                                                                                |
|  Welcome to the ASReview Automated Systematic Review software.                 |
|  In this mode the computer will simulate how well the ASReview software        |
|  could have accelerate the systematic review of your dataset.                  |
|  You can sit back and relax while the computer runs this simulation.           |
|                                                                                |
|  GitHub page:        https://github.com/asreview/asreview                      |
|  Questions/remarks:  asreview@uu.nl                                            |
|                                                                                |
---------------------------------------------------------------------------------

Traceback (most recent call last):
  File "/opt/homebrew/Caskroom/miniforge/base/envs/asreview_m1/bin/asreview", line 33, in <module>
    sys.exit(load_entry_point('asreview', 'console_scripts', 'asreview')())
  File "/Users/ottomattas/Documents/asreview/asreview/asreview/__main__.py", line 80, in main
    entry.load()().execute(sys.argv[2:])
  File "/Users/ottomattas/Documents/asreview/asreview/asreview/entry_points/simulate.py", line 48, in execute
    review_simulate(path, **args_dict)
  File "/Users/ottomattas/Documents/asreview/asreview/asreview/review/factory.py", line 314, in review_simulate
    review(dataset, *args, mode='simulate', **kwargs)
  File "/Users/ottomattas/Documents/asreview/asreview/asreview/review/factory.py", line 268, in review
    reviewer = get_reviewer(*args, mode=mode, model=model, **kwargs)
  File "/Users/ottomattas/Documents/asreview/asreview/asreview/review/factory.py", line 229, in get_reviewer
    reviewer = ReviewSimulate(as_data,
  File "/Users/ottomattas/Documents/asreview/asreview/asreview/review/simulate.py", line 89, in __init__
    raise ValueError("Expected fully labeled dataset.")
ValueError: Expected fully labeled dataset.
```

Furthermore, I ran a test on a fully labeled benchmark dataset [van_de_Schoot_2017.csv](https://github.com/asreview/systematic-review-datasets/blob/0909240fec2602c3ecef82d131790d390edb33bf/datasets/van_de_Schoot_2017/output/van_de_Schoot_2017.csv):
```
% asreview simulate van_de_Schoot_2017.csv --state_file myreview.h5                    

            _____ _____            _
     /\    / ____|  __ \          (_)
    /  \  | (___ | |__) |_____   ___  _____      __
   / /\ \  \___ \|  _  // _ \ \ / / |/ _ \ \ /\ / /
  / ____ \ ____) | | \ \  __/\ V /| |  __/\ V  V /
 /_/    \_\_____/|_|  \_\___| \_/ |_|\___| \_/\_/

---------------------------------------------------------------------------------
|                                                                                |
|  Welcome to the ASReview Automated Systematic Review software.                 |
|  In this mode the computer will simulate how well the ASReview software        |
|  could have accelerate the systematic review of your dataset.                  |
|  You can sit back and relax while the computer runs this simulation.           |
|                                                                                |
|  GitHub page:        https://github.com/asreview/asreview                      |
|  Questions/remarks:  asreview@uu.nl                                            |
|                                                                                |
---------------------------------------------------------------------------------

The following records are prior knowledge:

2616 - Six years after the wave. Trajectories of posttraumatic stress following a nat..   Johannesson, K. B., Arinell, H., Arnbe..
6075 - Post-traumatic stress disorder and functioning and quality of life outcomes in..   Zatzick, D. F., Weiss, D. S., Marmar, ..
```
And then removed one label manually, which gave the expected result:
```
% asreview simulate van_de_Schoot_2017.csv --state_file myreview.h5

            _____ _____            _
     /\    / ____|  __ \          (_)
    /  \  | (___ | |__) |_____   ___  _____      __
   / /\ \  \___ \|  _  // _ \ \ / / |/ _ \ \ /\ / /
  / ____ \ ____) | | \ \  __/\ V /| |  __/\ V  V /
 /_/    \_\_____/|_|  \_\___| \_/ |_|\___| \_/\_/

---------------------------------------------------------------------------------
|                                                                                |
|  Welcome to the ASReview Automated Systematic Review software.                 |
|  In this mode the computer will simulate how well the ASReview software        |
|  could have accelerate the systematic review of your dataset.                  |
|  You can sit back and relax while the computer runs this simulation.           |
|                                                                                |
|  GitHub page:        https://github.com/asreview/asreview                      |
|  Questions/remarks:  asreview@uu.nl                                            |
|                                                                                |
---------------------------------------------------------------------------------

Traceback (most recent call last):
  File "/opt/homebrew/Caskroom/miniforge/base/envs/asreview_m1/bin/asreview", line 33, in <module>
    sys.exit(load_entry_point('asreview', 'console_scripts', 'asreview')())
  File "/Users/ottomattas/Documents/asreview/asreview/asreview/__main__.py", line 80, in main
    entry.load()().execute(sys.argv[2:])
  File "/Users/ottomattas/Documents/asreview/asreview/asreview/entry_points/simulate.py", line 48, in execute
    review_simulate(path, **args_dict)
  File "/Users/ottomattas/Documents/asreview/asreview/asreview/review/factory.py", line 314, in review_simulate
    review(dataset, *args, mode='simulate', **kwargs)
  File "/Users/ottomattas/Documents/asreview/asreview/asreview/review/factory.py", line 268, in review
    reviewer = get_reviewer(*args, mode=mode, model=model, **kwargs)
  File "/Users/ottomattas/Documents/asreview/asreview/asreview/review/factory.py", line 229, in get_reviewer
    reviewer = ReviewSimulate(as_data,
  File "/Users/ottomattas/Documents/asreview/asreview/asreview/review/simulate.py", line 89, in __init__
    raise ValueError("Expected fully labeled dataset.")
ValueError: Expected fully labeled dataset.
```

**Comment by ottomattas** [comment]
> No @ottomattas, there is no issue yet. maybe good to create one. There is a comment of a TensorFlow dev here [#711 (comment)](https://github.com/asreview/asreview/pull/711#issuecomment-912823615).

I see that the Tensorflow PR has been merged and our test are also running successfully now. I guess, the issue is not needed any longer? @J535D165

File: tests/test_simulate.py (hunk 1 above)
Comment (J535D165) [reply tests/test_simulate.py:216]: Oke looks good.

**Comment by J535D165** [comment]
Thanks. Happy this feature is removed (for now). 

To the users of this feature: Please remove the unlabeled rows from your data in a spreadsheet editor (like Excel) or programming language. The removal of this function makes ASReview simulate more reliable at the moment.

**Pull request merged**